pkg/pf/tests/tfgen_test.go

```diff
@@ -63,3 +63,36 @@ func TestRenameResourceWithAliasInAugmentedProvider(t *testing.T) {
 	_, err = i.P.(*muxer.ProviderShim).ResolveDispatch(&i)
 	require.NoError(t, err)
 }
+
+func TestRenameDataSourceWithAliasInAugmentedProvider(t *testing.T) {
```

Comment:
I think we need a test that shows that old and new dispatch actually resolve correctly, (from the Pulumi level), not just that the muxer doesn't fail to resolve.

Also, the text only applies to muxed datasources, so lets put that in the title.
```suggestion
func TestRenameMuxedDataSourceWithAliasInAugmentedProvider(t *testing.T) {
```